specs/anchored-region/anchored-region.md

```diff
@@ -0,0 +1,254 @@
+    </anchored-region>
+    ...stuff...
+</div>
+
+```
+
+A region that renders above or below the anchor depending on available space but only on initial layout.
+```
+<div id="viewport">
+    ...stuff...
+    <button id="anchor">
+        Button is an anchor
+    </button>
+    <anchored-region
+        anchorid="anchor"
+        verticalpositioningmode="onetime"
+    >
+      This shows up above or below the anchor depending on available space
+    </anchored-region>
+    ...stuff...
+</div>
+
+```
+
+A region that overlaps the anchor and renders above or below it depending on available space.
+
+```
+<div id="viewport">
+    ...stuff...
+    <button id="anchor">
+        Button is an anchor
+    </button>
+    <anchored-region
+        anchorid="anchor"
+        verticalPositioningMode="dynamic"
+        verticalinset="true"
+    >
+      This overlaps the anchor and extends above or below it depending on available space.
+    </anchored-region>
+    ...stuff...
+</div>
+
+```
+
+A region renders above or below the anchor based on available space and is sized to match the available.
+
+```
+<div id="viewport">
+    ...stuff...
+    <button id="anchor">
+        Button is an anchor
+    </button>
+    <anchored-region
+        anchorid="anchor"
+        verticalPositioningMode="dynamic"
+        verticalscalingenabled="true"
+    >
+       This region renders above or below the anchor based on available space and sizes itself to match.
+    </anchored-region>
+    ...stuff...
+</div>
+
+```
+
+A region that renders below the anchor until that space is less than 100px.
+```
+<div id="viewport">
+    ...stuff...
+    <button id="anchor">
+        Button is an anchor
+    </button>
+    <anchored-region
+        anchorid="anchor"
+        verticalpositioningmode="dynamic"
+        verticaldefaultposition="below"
+        verticalthreshold="100"
+    >
+      This shows renders below the anchor as long as there is at least 100px available there.
+    </anchored-region>
+    ...stuff...
+</div>
+```
+
+The dimensions of the anchored region will match the dimensions of the content unless scaling is enabled on a particular axis (verticalscalingenabled & horizontalscalingenabled) in which case it will fill all available space between the anchor and viewport.
+
+### API
+
+*Component name:*
+- `fast-anchored-region`
+
+*Attributes:*
+- anchorId - The html id of the HTMLElement used as the anchor around which the positioning region is placed.  This must be set for the component's positioning logic to be active.
+
+- viewportId - The html id of the HTMLElement used as the viewport the component uses to determine available layout space around the anchor element.  If unset the immediate parent element of the positioner is used.
+
+- horizontalPositioningMode - Can be 'uncontrolled', 'locktodefault', 'dynamic' or 'onetime'.  Default is 'uncontrolled'.
+- horizontalDefaultPosition - Can be 'left', 'right' or 'unset'.  Default is 'unset'
```

Comment:
do we *also* want to support "start" and "end" which adjust depending on document-direction, similarly to the `*-inline-start', `*-inline-end`, and flex properties?

Reply:
I don't think so.  Authors should be able to apply custom css the scaling region and/or its internals to achieve whatever layout logic they want to the region contents.

Reply:
I mean to control the default position, not the internals. The case is when I want the `horizontalDefaultPosition` to be "left" in LTR but "right" in RTL

Reply:
Oh, sorry, yeah I think we could do that.  I will add it.  Do we have somewhere to put component shared resources?  I think we'll want to put a getCurrentDirection() like thing there.

Reply:
I think that can go in the `@microsoft/fast-web-utilities` package